tests/main.test.ts

```diff
@@ -231,6 +258,82 @@ it(`should ignore the packageManager field when found within a node_modules vend
   });
 });
 
+it(`should use hash from "packageManager" even when "devEngines" defines a different one`, async () => {
```

Comment:
Shouldn’t this error?

Reply:
I don’t think so, according to semver spec, two versions with same number and different build metadata should refer to the same release. In https://github.com/nodejs/corepack/issues/316 someone requested to have the possibility to put build metadata that’s not a hash, so I figured we could use devEngines for that

Reply:
> two versions with same number and different build metadata

So like one has a sha1 hash and the other a sha512 hash, something like that? I was thinking of two hashes of the same type but different values, and I was wondering how that could ever be something other than an error (since only one could be correct).

Reply:
I'm using that part of the semver spec as reference:

> Build metadata MUST be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence. Examples: 1.0.0-alpha+001, 1.0.0+20130313144700, 1.0.0-beta+exp.sha.5114f85, 1.0.0+21AF26D3----117B344092BD.

My understanding is that it doesn't matter whether the build metadata refer to a hash or not, as long as it has the same version number, it refers to the same version, and therefor should pass validation – as long as the hash in `packageManager` corresponds to the download archive.